File: Data Structures & Algorithms/duplicate-integer/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        let m = {};

        for(let i = 0; i < nums.length; i++) {
            m[nums[i]] = (m[nums[i]] ?? 0) + 1;
        }

        for(const key in m) {
            if(m[key] > 1) {
                return true
            }
        }

        return false;
    }
}

File: Data Structures & Algorithms/subsets/submission-4.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []

        def backtrack(i, sub):
            if i >= len(nums):
                print(sub, i)
                res.append(sub.copy())
                return
            sub.append(nums[i])
            backtrack(i + 1, sub)
            sub.pop()
            backtrack(i + 1, sub)

        backtrack(0, [])

        return res
        